include/seastar/core/coroutine.hh

```diff
@@ -30,6 +30,7 @@
 
 #ifndef SEASTAR_MODULE
 #include <coroutine>
+#include <source_location>
```

Comment:
it took me a while to check if it's safe to use `<source_location>` instead of `util/std-compat.hh` . the answer is yes. as we support the latest two major versions of the GCC and Clang compilers. in general, we use libstdc++, which added the support to `source_location` in https://github.com/gcc-mirror/gcc/commit/57d76ee9cf6265e012fad6286adfaeaba9414c11, which is included by gcc 12 and up. and the latest stable release of GCC is GCC 14.2.

Reply:
Opened https://github.com/scylladb/seastar/issues/2961 on this.